uit/uit.py

```diff
@@ -726,6 +744,92 @@ def _parse_hpc_output(cls, output, as_df, columns=None, delimiter=None, delimite
+
+        debug_header = f" {FG_RED}time={time_text}{ALL_OFF}    node={self.login_node}"
+
+        if len(local_vars['r'].text) > 0:
+            resp = local_vars['r'].json()
+        else:
+            resp = {}
+
+        if resp.get('exitcode') is not None:
+            debug_header += f"    rc={resp.get('exitcode')}"
+        debug_header += f"    username={self.username}"
+
+        # stdout and stderr will only show up for call() and if they contain text
+        nice_stdout = ""
+        if resp.get('stdout'):
+            nice_stdout = "\n  stdout="
+            if len(resp.get('stdout')) > 500:
+                nice_stdout += "'" + resp.get('stdout')[:500].replace('\n', '\\n') + \
+                              f"'  <len:{len(resp.get('stdout'))}>"
+            else:
+                nice_stdout += "'" + resp.get('stdout').replace('\n', '\\n') + "'"
+
+        nice_stderr = ""
+        if resp.get('stderr'):
+            nice_stderr = "\n  stderr="
+            if len(resp.get('stderr')) > 500:
+                nice_stderr += "'" + resp.get('stderr')[:500].replace('\n', '\\n') + \
+                              f"'  <len:{len(resp.get('stderr'))}>"
+            else:
+                nice_stderr += "'" + resp.get('stderr').replace('\n', '\\n') + "'"
+
+        # Show only relevant function calls and ignore standard library
+        only_show_stacktrace = ['pyuit',
+                                'helios',
+                                'tethysapp',
+                                'uit_plus',
+                                'kestrel',
+                                'galaxy',
+                                'Galaxy']
+        stacktrace = traceback.extract_stack()
+        nice_trace = ''
+        for i in range(0, len(stacktrace)):
+            if stacktrace[i].name == 'wrapper' or stacktrace[i].name == 'wrap_f':
+                # ignore the decorators for call()
+                continue
+            if 'traceback.extract_stack()' in stacktrace[i].line:  # ignore this last line
+                continue
+            if 'self._debug_uit(' in stacktrace[i].line:  # ignore this function call
+                continue
+            for substring in only_show_stacktrace:
+                if substring in stacktrace[i].filename:
+                    # Simple approach: grab the last 3 folders
+                    trimmed_filename = os.sep.join(stacktrace[i].filename.split(os.sep)[-4:])
+                    # Nicer approach: try to display only the folders that start with pyuit or helios, etc.
+                    for j, path_element in enumerate(stacktrace[i].filename.split(os.sep)):
+                        if substring in path_element:
+                            trimmed_filename = os.sep.join(stacktrace[i].filename.split(os.sep)[j:])
+                            break
+                    nice_trace += f"\n    {i}: {trimmed_filename}:" + \
+                                  f"{stacktrace[i].lineno} {stacktrace[i].name}()" + \
```

Comment:
I'm not overly concerned about the stickler errors here, but since you are making some changes anyway, maybe consider something like this?

```python
 nice_trace += (
    f"\n    {i}: {trimmed_filename}:"
    f"{stacktrace[i].lineno} {stacktrace[i].name}()"
    f"    {stacktrace[i].line.encode('ascii', 'backslashreplace').decode('ascii')}"
)
```

Reply:
I learned that PEP8 prefers your code style. Fixed.